components/client-attestation-mgt/org.wso2.carbon.identity.client.attestation.mgt/src/main/java/org/wso2/carbon/identity/client/attestation/mgt/validators/AndroidAttestationValidator.java

```diff
@@ -0,0 +1,289 @@
+     * @return True if the request details are valid; false otherwise.
+     */
+    private boolean validateRequestDetails(DecodeIntegrityTokenResponse decodeIntegrityTokenResponse,
+                                           ClientAttestationContext clientAttestationContext) {
+
+        // Extract request details from the response.
+        RequestDetails requestDetails = decodeIntegrityTokenResponse.getTokenPayloadExternal().getRequestDetails();
+
+        String expectedPackageName = clientAttestationMetaData.getAndroidPackageName();
+
+        // Get the current time in milliseconds.
+        long currentTimeInMillis = Calendar.getInstance(TimeZone.getTimeZone(UTC)).getTimeInMillis();
+
+        // Get the request time from the token response.
+        long requestTimeInMillis = requestDetails.getTimestampMillis();
+
+        long allowedWindowMillis;
+
+        String allowedWindow = IdentityUtil.getProperty(CLIENT_ATTESTATION_ALLOWED_WINDOW_IN_MILL_SECOND);
+
+
+        if (!StringUtils.equals(requestDetails.getRequestPackageName(), expectedPackageName)) {
+            // The package name in the request details does not match the requested client.
+            clientAttestationContext.setAttested(false);
+            clientAttestationContext.setValidationFailureMessage("Package name in the request details does " +
+                    "not match with the requested client.");
+            return false;
+        } else if (StringUtils.isNotEmpty(allowedWindow)) {
+            try {
+                allowedWindowMillis = Long.parseLong(allowedWindow);
+                if (currentTimeInMillis - requestTimeInMillis > allowedWindowMillis) {
+                    // The generated Integrity token is considered old, likely due to a replay attack.
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Attestation request provided by Client :" + applicationResourceId +
+                                " in tenant : " + tenantDomain + " is older than required window.");
+                    }
+                    clientAttestationContext.setAttested(false);
+                    clientAttestationContext.setValidationFailureMessage("Attestation request provided by Client " +
+                            "is older than required window.");
+                    return false;
+                } else {
+                    // Request details are valid.
+                    return true;
+                }
+            } catch (NumberFormatException e) {
+                LOG.error("Error while parsing attestation allowed window timeout config: " + allowedWindow, e);
+                clientAttestationContext.setAttested(false);
+                clientAttestationContext.setValidationFailureMessage("Error while parsing attestation allowed window " +
+                        "timeout config. Probably a misconfiguration, hence rejecting the request.");
+                return false;
```

Comment:
Lets fallback to a hardcoded default or throw the exception out and break the flow.